dotcom-rendering/src/components/AdPortals.importable.tsx

```diff
@@ -0,0 +1,178 @@
+import { AdSlot as BridgetAdSlot } from '@guardian/bridget/AdSlot';
+import type { IRect as BridgetRect } from '@guardian/bridget/Rect';
+import { isNonNullable } from '@guardian/libs';
+import libDebounce from 'lodash.debounce';
+import type { RefObject } from 'react';
+import { createRef, useEffect, useRef, useState } from 'react';
+import { createPortal } from 'react-dom';
+import { getCommercialClient } from '../lib/bridgetApi';
+import { AdSlot } from './AdSlot.apps';
+
+const calculateAdPosition = (element: Element): BridgetRect => {
+	const elementRect = element.getBoundingClientRect();
+
+	const scrollLeft =
+		document.scrollingElement !== null
+			? document.scrollingElement.scrollLeft
+			: document.body.scrollLeft;
+	const scrollTop =
+		document.scrollingElement !== null
+			? document.scrollingElement.scrollTop
+			: document.body.scrollTop;
+
+	// Potential optimisation: round these numbers to be less specific to reduce bridget calls to update ad positions.
+	return {
+		x: elementRect.left + scrollLeft,
+		y: elementRect.top + scrollTop,
+		width: elementRect.width,
+		height: elementRect.height,
+	};
+};
+
+const positionsEqual = (a: BridgetRect, b: BridgetRect): boolean =>
+	a.height === b.height && a.width === b.width && a.x === b.x && a.y === b.y;
+
+const getIsPremium = () => Promise.resolve(false);
```

Comment:
This will be replaced with a bridget implementation in [this ticket](https://github.com/guardian/dotcom-rendering/issues/8804) which is currently in progress